graph2tac/tfgnn/tasks.py

```diff
@@ -615,6 +646,228 @@ def metrics() -> Dict[str, List[tf.keras.metrics.Metric]]:
         return {TacticPrediction.TACTIC_LOGITS: [tf.keras.metrics.SparseCategoricalAccuracy(name="accuracy")]}
 
 
+class GlobalEmbeddings(tf.keras.layers.Layer):
+    def __init__(
+        self,
+        global_embeddings_layer: LogitsFromEmbeddings,
+        dynamic_global_context: bool,
+        name="global_embeddings",
+        **kwargs
+    ):
+        super().__init__(name=name, **kwargs)
+        self.global_embeddings_layer = global_embeddings_layer
+        self.dynamic_global_context = dynamic_global_context
+
+    def call(
+        self, 
+        global_context: tf.RaggedTensor,  # [batch, None(context)]
+        inference: bool = False,
+        training=False
+    ) -> tf.Tensor: # [batch, None(context), hdim]
+        # [valid_context, hdim]
+        all_embeddings = self.global_embeddings_layer.get_keys_embeddings()
+
+        # note: inference model always uses dynamic global context
+        if self.dynamic_global_context or inference:
```

Comment:
Is there still any meaning in `self.dynamic_global_context = False`? Perhaps, we could discard this option.

Reply:
We could still support it.  We would need to convert the full global context to the dynamic one right before the output, but for training purposes it would do the multiplication on all the definitions.  But also, I don't see the point, so maybe we should just kill this option to simplify the code.  (It is broken currently, anyway.)